src/logging.c

```diff
@@ -367,17 +366,27 @@ void log_print(int lvl, const char* format, ...) {
  * Log the message with the given level. New line character
  *              is appended to the error stream.
  * @param lvl     level of the message
+ * @param file    filename where the log occurred
+ * @param line    line number in the file
+ * @param lvl     level of the message
  * @param format  format of the message
  *            ... - the additional arguments
  */
 
-void log_println(int lvl, const char* format, ...) {
+void log_println_impl(int lvl, const char* file, int line, const char* format, ...) {
```

Comment:
I have a nagging feeling that these two functions could be combined or somehow simpler. But I cannot quantify it, which makes me thing that it probably wouldn't be quick. It's not apparent to me what additional value the I2ErrLogVT functions (or I2ErrHandles) provide here over direct implementation.